src/deepcell/data_splitter.py

```diff
@@ -65,33 +71,143 @@ def __init__(self, model_inputs: List[ModelInput], train_transform=None,
         self._centroid_use_mask = centroid_use_mask
 
         if center_roi_centroid not in ('test', True, False):
-            raise ValueError(f'Invalid value for center_soma. Valid '
-                             f'values are "test", True, or False')
+            raise ValueError('Invalid value for center_soma. Valid '
+                             'values are "test", True, or False')
         self._center_roi_centroid = center_roi_centroid
 
-    def get_train_test_split(self, test_size):
-        full_dataset = RoiDataset(
-            model_inputs=self._model_inputs,
-            cre_line=self.cre_line,
-            mask_out_projections=self.mask_out_projections,
-            image_dim=self.image_dim,
-            use_correlation_projection=self._use_correlation_projection
-        )
-        sss = StratifiedShuffleSplit(n_splits=2, test_size=test_size,
+    @staticmethod
+    def _get_experiment_groups_for_stratified_split(
+            experiment_metadatas: List[ExperimentMetadata],
+            exp_ids: np.ndarray,
+            n_depth_bins: int):
+        """Bin up input experiment data into bins of depth, experiment type,
+        and "problem" experiments.
+
+        Parameters
+        ---------
+        experiment_metadatas : List[ExperimentMetadata]
+            Set of experiment metadatas to stratify.
+        exp_ids : np.ndarray
+            Array of input experiment ids. Must have same order as
+            ``experiment_metadatas``.
+        n_depth_bins
+            Number of of bins to split by experiment depth. Bins are
+            selected to have an equal number of experiments in each bin.
+
+        Returns
+        -------
+        exp_bin_ids : numpy.ndarray
+           Bin assigned to each experiment id.
+        """
+        # Create our arrays we will use to bin. -1 for a bin_id denotes
+        # a experiment that has not been assigned to a bit yet.
+        exp_bin_ids = np.full(len(experiment_metadatas), -1, dtype=int)
+        depths = np.empty(len(experiment_metadatas), dtype=int)
+
+        for idx, exp_meta in enumerate(experiment_metadatas):
+            if exp_meta['problem_experiment']:
```

Comment:
Understandable. Added this comment block.
`# Check for any "special" experiments that we want to partition
  # separately. This can be anything from labeling a set that have
  # problematic segmentation or have some other special attribute
  # we would like to sample fairly in the train/test split.`